packages/docusaurus-theme-common/src/index.ts

```diff
@@ -19,147 +23,54 @@ export {
+export {useCurrentSidebarCategory} from './utils/docsUtils';
 
 export {usePluralForm} from './utils/usePluralForm';
 
-export {useLocationChange} from './utils/useLocationChange';
-
 export {useCollapsible, Collapsible} from './components/Collapsible';
 
+export {ThemeClassNames} from './utils/ThemeClassNames';
+
 export {
-  useDocsPreferredVersion,
-  useDocsPreferredVersionByPluginId,
-  DocsPreferredVersionContextProvider,
-} from './contexts/docsPreferredVersion';
+  useIsomorphicLayoutEffect,
+  useDynamicCallback, // TODO rename to useEvent()
+  usePrevious,
+  ReactContextError,
+} from './utils/reactUtils';
 
-export {duplicates, uniq} from './utils/jsUtils';
+export {PageMetadata, HtmlClassNameProvider} from './utils/metadataUtils';
 
-export {ThemeClassNames} from './utils/ThemeClassNames';
+export {useColorMode, type ColorMode} from './contexts/colorMode';
 
 export {
-  AnnouncementBarProvider,
-  useAnnouncementBar,
-} from './contexts/announcementBar';
+  NavbarSecondaryMenuFiller,
+  type NavbarSecondaryMenuComponent,
+} from './contexts/navbarSecondaryMenu/content';
 
-export {useLocalPathname} from './utils/useLocalPathname';
+export {useWindowSize} from './hooks/useWindowSize';
+
+/*
+ * APIs kept undocumented, on purpose
+ * Note: we still guarantee retro-compatibility on those
+ */
 
 export {
   translateTagsPageTitle,
   listTagsByLetters,
   type TagLetterEntry,
 } from './utils/tagsUtils';
 
-export {useHistoryPopHandler} from './utils/historyUtils';
-
-export {
-  useTOCHighlight,
-  type TOCHighlightConfig,
-} from './hooks/useTOCHighlight';
-
-export {
-  useFilteredAndTreeifiedTOC,
-  useTreeifiedTOC,
-  type TOCTreeNode,
-} from './utils/tocUtils';
-
 export {isMultiColumnFooterLinks} from './utils/footerUtils';
 
-export {
-  ScrollControllerProvider,
-  useScrollController,
-  useScrollPosition,
-  useScrollPositionBlocker,
-  useSmoothScrollTo,
-} from './utils/scrollUtils';
-
-export {
-  useIsomorphicLayoutEffect,
-  useDynamicCallback,
-  usePrevious,
-  ReactContextError,
-} from './utils/reactUtils';
-
 export {isRegexpStringMatch} from './utils/regexpUtils';
 
-export {useHomePageRoute, isSamePath} from './utils/routesUtils';
-
-export {
-  PageMetadata,
-  HtmlClassNameProvider,
-  PluginHtmlClassNameProvider,
-} from './utils/metadataUtils';
-
-export {
-  useColorMode,
-  ColorModeProvider,
-  type ColorMode,
-} from './contexts/colorMode';
-
-export {splitNavbarItems, NavbarProvider} from './utils/navbarUtils';
-
-export {
-  useTabGroupChoice,
-  TabGroupChoiceProvider,
-} from './contexts/tabGroupChoice';
-
-export {useNavbarMobileSidebar} from './contexts/navbarMobileSidebar';
-export {
-  NavbarSecondaryMenuFiller,
-  type NavbarSecondaryMenuComponent,
-} from './contexts/navbarSecondaryMenu/content';
-export {useNavbarSecondaryMenu} from './contexts/navbarSecondaryMenu/display';
+export {duplicates, uniq} from './utils/jsUtils';
 
 export {useBackToTopButton} from './hooks/useBackToTopButton';
```

Comment:
Is this really stable? We have a lot of "wiring hooks" like `useAnnouncementBar`, `useNavbarMobileSidebar`, etc., and this is the only public one.